### ISSUE TITLE
[Bug]: `AXManualAccessibility` attribute can't be set (`kAXErrorAttributeUnsupported`)

### ISSUE THREAD

**Issue by Rhys-T** [electron/electron#37465]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

22.0.1 (Electron Fiddle), 22.0.2 (demo app launched from Fiddle), 23.1.0 (Element), 19.1.9 (VS Code)

### What operating system are you using?

macOS

### Operating System Version

macOS Catalina 10.15.7

### What arch are you using?

x64

### Last Known Working Electron version

unknown

### Expected Behavior

As described [in the Electron docs](https://www.electronjs.org/docs/latest/tutorial/accessibility/#macos), it should be possible to set the `AXManualAccessibility` attribute[^axma] of an Electron-based application through the Accessibility API, to force Electron/Chromium to build the full tree of accessibility info.

[^axma]: Added in #10305. The private `AXEnhancedUserInterface` attribute (used by VoiceOver) also causes Electron/Chromium to build the full a11y tree, but has some unwanted side effects.

### Actual Behavior

Attempting to set `AXManualAccessibility` always results in an error (`kAXErrorAttributeUnsupported` or its equivalent), as if the attribute doesn't exist. I can replicate this in Swift, AppleScript (GUI scripting through the System Events app), and [Hammerspoon](https://github.com/Hammerspoon/hammerspoon) (through [the `hs.axuielement` module](https://www.hammerspoon.org/docs/hs.axuielement.html)). UIElementInspector also doesn't list it as an existing attribute.

### Additional Information

I believe this is a continuation of #30644, which was auto-closed as stale.

I haven't included a test case gist because this happens with _any_ Electron app I try - from the default code that Fiddle opens with, all the way up to VS Code.

I can see in electron_application.mm [where it checks if that attribute is what's being set](https://github.com/electron/electron/blob/9b20b3a722a7b807c161813cf2b616e74aee660d/shell/browser/mac/electron_application.mm#L182), but I don't see any code that tells the Accessibility API that that attribute actually _exists_ to be set. It looks to me like fixing that would involve overriding [the `accessibilityAttributeNames` method](https://developer.apple.com/documentation/objectivec/nsobject/1525181-accessibilityattributenames?language=objc) ([example here](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Accessibility/cocoaAXSupportingAttributes/cocoaAXSupportAttributes.html)). That method is from a deprecated informal protocol that suggests the `NSAccessibility` protocol instead, but:

a) `NSAccessibility` doesn't seem to have any replacement for `accessibilityAttributeNames` - the new protocol doesn't seem to support custom attributes (or custom anything, really).
b) `accessibilitySetValue:forAttribute:` - which Electron is already overriding - is also part of that deprecated informal protocol, and also doesn't seem to have a replacement.

#### Examples of various ways I've tried setting the attribute:

<details>
<summary>Swift example</summary>

```swift
#!/usr/bin/swift
import Cocoa
let name = CommandLine.arguments.count >= 2 ? CommandLine.arguments[1] : "Electron"
let pid = NSWorkspace.shared.runningApplications.first(where: {$0.localizedName == name})!.processIdentifier
let axApp = AXUIElementCreateApplication(pid)
let error = AXUIElementSetAttributeValue(axApp, "AXManualAccessibility" as CFString, true as CFTypeRef)
dump(error.rawValue)
// (couldn't get Swift to return the error value's name outside the REPL, so I'm using the rawValue)
// should be 0
// actually returns -25205 (AXError.attributeUnsupported)
```

</details>

<details>
<summary>AppleScript example</summary>

```applescript
#!/usr/bin/osascript

-- Note: you'll probably have to run this from Script Editor.
-- Running in the terminal will give the error "Not authorized to send Apple events to System Events."

on run argv
	if (count argv) ≥ 1 then
		set appName to first item of argv
	else
		set appName to "Electron"
	end if
	tell application "System Events"
		tell application process appName
			set value of attribute "AXManualAccessibility" to true
		end tell
	end tell
end run
```

</details>

<details>
<summary>Hammerspoon (Lua) example</summary>

```lua
-- paste this into Hammerspoon console
-- (all at once - if you run the lines separately, the locals will disappear between lines)
local appName = 'Electron' -- or whatever
local app = hs.application.get(appName)
local axApp = hs.axuielement.applicationElement(app)
return axApp:setAttributeValue('AXManualAccessibility', true)
-- returns:
-- 	nil	Attribute is not supported by target
-- you can also do:
-- 	axApp.AXManualAccessibility = true
-- but that just fails silently
```

</details>

**Comment by codebytere** [comment]
hey @Rhys-T it took me a bit but i just popped up a PR to fix this!
